lib/handlers/subscribe.js

```diff
@@ -60,6 +60,21 @@ function storeSubscriptions (sub, done) {
   var broker = client.broker
   var perst = broker.persistence
 
+  if (sub && packet.subscriptions[0].topic !== sub.topic) {
+    // don't call addSubscriptions per topic,
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    return done(null, sub)
+  }
+
+  if (!packet.messageId) {
+    // this is a restoreSubscription, don't addSubscriptions!
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    if (perst._addedSubscriptions) {
+      perst._addedSubscriptions(client, packet.subscriptions)
+    }
```

Comment:
It's not 100% clear to me how the code with `subscriptionRestored` and `subscriptionAdded` would play, but yes, go ahead. Any reference here to `_addedSubscriptions`  should disappear.